### ISSUE TITLE
Size matters - scala.reflect.macros.TypecheckException within an assert-expression when a property with the name 'size' of type Float is being accessed

### ISSUE THREAD

**Issue by FloWi** [scalatest/scalatest#478]
I found a very strange bug.
If have a case-class with a property named `size` of type `Float`.
When I access this property within an assert-expression, the compile of the testcase fails with following exception
```
[error] scala.reflect.macros.TypecheckException: type mismatch;
[error]  found   : Float 
[error]  required: Long
```

A colleague of mine suggested (by chance) to change the name of the property. And it works fine. 
So size does matter ;-)

Any idea? Is the macro somehow expecting a size-property to be of type Long, because of the size-property of all the collection-classes? Just a guess...

Regards,
Florian

Below is a mini-example of the weird behaviour.

```scala
import org.scalatest.{ShouldMatchers, BeforeAndAfterAll, FunSuite}

class HelloSpec extends FunSuite
with BeforeAndAfterAll
with ShouldMatchers {

  /* comment this testcase out to get the compile working  */
  test("Size matters - within assert-expression fails to compile") {

    case class TestObject(size: Float)

    val testObject: TestObject = TestObject(size = 42f)

    /*
      [error] /home/flwi/dev/playground/scalatest-bug-TypecheckException/src/test/scala/HelloSpec.scala:20: exception during macro expansion:
      [error] scala.reflect.macros.TypecheckException: type mismatch;
      [error]  found   : Float
      [error]  required: Long
     */
    assert(testObject.size == 42f)
  }

  test("Size matters - works fine when extracted as a variable") {

    case class TestObject(size: Float)

    val testObject: TestObject = TestObject(size = 42f)
    val size: Float = testObject.size
    assert(size == 42f)
  }

  test("bigger is better") {

    case class TestObject(big: Float)

    val testObject: TestObject = TestObject(big = 42f)
    assert(testObject.big == 42f)
  }
}
```

**Comment by rgitzel** [comment]
We just ran up against the same head-scratcher, only using "length":
```
  test("length also matters") {

    case class TestObject(length: Float)

    val testObject: TestObject = TestObject(length = 42f)
    assert(testObject.length == 42f)
  }
```

Very strange, indeed!   Particular since it showed up only when we upgraded ScalaTest on an older repo.  We thought we'd somehow screwed up a configuration somewhere.   It was a complete surprise to learn it was triggered by our code, ultimately.

**Comment by cheeseng** [comment]
@FloWi @rgitzel The newly released 2.2.6-SNAP1 should fix this problem, do you mind giving it a try?

**Comment by bvenners** [comment]
This was fixed several months ago but we never closed the issue. Closing now.
